### PR TITLE
[LS] Code action framework and order entities code action.

### PR DESCRIPTION
**Pull request by gdotdesign**
This pull request adds the framework for handling [code actions] in the language server and a code action to reorder the entities (constants and functions) in a module. I added this first because I'm going through the core library and didn't want to manually reorder them :joy:

### CODE REVIEW

File: src/ast/node.cr [hunk 1]
```diff
@@ -40,8 +40,8 @@ module Mint
         end
       end
 
+      property from : Int32
```
Comment (gdotdesign) [inline src/ast/node.cr:43]: This needed to change because it's used for the reordering.

**Review by Sija — APPROVED** [review]
LGTM

File: src/ast/node.cr (hunk 1 above)
Comment (Sija) [reply src/ast/node.cr:43]: I'm not following why this needs to be declared earlier?

Comment (gdotdesign) [reply src/ast/node.cr:43]: Not earlier, it changed from a `getter` to a `property`.

Comment (gdotdesign) [reply src/ast/node.cr:43]: It's earlier because of my slight OCD of having longer lines at the top :joy:

**Pull request merged**